src/daft-core/src/array/ops/list.rs

```diff
@@ -118,6 +118,61 @@ pub fn get_slices_helper(
     .into_series())
 }
 
+fn get_chunks_helper(
+    to_skip: Option<impl Iterator<Item = usize>>,
+    total_elements_to_skip: usize,
+    flat_child: &Series,
+    field: Arc<Field>,
+    new_offsets: Vec<i64>,
+    validity: Option<&arrow2::bitmap::Bitmap>,
+    size: usize,
+) -> DaftResult<Series> {
+    // If there is a list where the number of elements that do not fit properly when chunked into
+    // sublists of `size` elements, we skip the elements that don't fit. If all lists in the
+    // Series can be chunked cleanly, we use a fast path where we pass the underlying array of
+    // elements to the result, but reinterpret it as a list of sublists. If there is at least one
+    // list that cannot be chunked cleanly, we use a slower path that removes excess elements.
+    if total_elements_to_skip == 0 {
+        let inner_list_field = field.to_exploded_field()?.to_fixed_size_list_field(size)?;
+        let inner_list = FixedSizeListArray::new(
+            inner_list_field.clone(),
+            Series::from_arrow(field.to_exploded_field()?.into(), flat_child.to_arrow())?,
```

Comment:
Why is this conversion to and from arrow necessary? I think `to_arrow` is meant mostly for FFI back into Python.

I think `flat_child.into_series()` should give you what you want here (you'll have to pull in the `IntoSeries` trait)

Reply:
Ah, you're right there was no need to make this so convoluted.

So there's an option of doing `flat_child.inner.into_series()` which goes from `&Series  -> SeriesLike -> Series`. In the end of the day I settled on `flat_child.clone()`, since `flat_child` is a Series which only contains `Arc<dyn SeriesLike>`, so only the Arc would be cloned.

Let me know if you think I should do otherwise.